OpenRA.Mods.Common/Traits/Player/MissionObjectives.cs

```diff
@@ -92,7 +94,7 @@ public int Add(Player player, string description, ObjectiveType type = Objective
 			return newID;
 		}
 
-		public void MarkCompleted(Player player, int objectiveID)
+		public void MarkCompleted(Player player, int objectiveID, string triggeredFlag = null, string finishedMission = null)
```

Comment:
These changes can then go away, replaced with the mission scripts individually calling a new Lua API to set the flags.